build/ci_utils/src/archive/extract_files.rs

```diff
@@ -9,5 +9,6 @@ pub trait ExtractFiles {
     /// IMPORTANT: If the function uses its input path to generate an output path, care must be
     /// taken that the output path is not in an unexpected location, especially if coming from an
     /// untrusted archive.
+    #[allow(async_fn_in_trait)]
```

Comment:
If we allow something, there's should be a comment explaining why lint is not right here.